File: mall-sellergoods-interface/src/main/java/com/mall/sellergoods/service/BrandService.java
package com.mall.sellergoods.service;

import java.util.List;

import com.mall.pojo.Brand;

import common.pojo.PageResult;

/**
 * 品牌接口
 * @author Wwl
 *
 */
public interface BrandService {

	/**
	 * 增加
	 * */
	public void add(Brand brand);

	/**
	 * 更新
	 * */
	public void update(Brand brand);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public Brand findOne(Long id);
	
	public List<Brand> findAll();
	
	/**
	 * 分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);

	/**
	 * 分页
	 * @param brand 品牌
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Brand brand,int pageNum,int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

}

File: mall-manager-web/src/main/webapp/js/service/brandService.js
/**
 * 品牌服务层
 * Created by Wwl on 2019/3/4
 */
app.service('brandService', function($http){

    //增加
    this.add = (entity) => $http.post('../brand/add.do',entity);

    //修改
    this.update = (entity) => $http.post('../brand/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../brand/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../brand/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../brand/list.do');

    //分页查询
    this.findPage = (page,size) => $http.get('../brand/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.post('../brand/search.do?page='+page+'&size='+size,searchEntity);

    //下拉列表数据
    this.findOptionList = () => $http.get('../brand/findOptionList.do');

});